README.md

```diff
@@ -213,6 +315,163 @@ module Maintenance
 end
 ```
 
+### Writing test for a custom Task
+
+As with other tasks, you should write tests for your `#process` method. It will
+receive the first item in each pair your custom Enumerator yields (`[item,
+item_cursor]`).
+
+You should also ensure your Enumerator is tested, by unit testing it in
+isolation, testing your `#enumerator_builder` method, or both. Make sure you
+test how your Enumerator handles the absence or presence of a `context.cursor`,
+if applicable.
+
+```ruby
+# test/tasks/maintenance/shopping_list_task_test.rb
+module Maintenance
+  class ShoppingListTaskTest < ActiveSupport::TestCase
+    test '#process adds ingredients to the shopping list' do
+      ingredient = recipes(:tacos).first
+      ShoppingList.expects(:add).with(ingredient)
+
+      ShoppingListTask.process(ingredient)
+    end
+
+    test '#enumerator_builder.enumerator enumerates ingredients for a random recipe' do
+      FancyRecipeAPI.fake_it_till_you_make_it do
+        recipe = recipes(:tacos)
+        FancyRecipeAPI.expects(:random_recipe).returns(recipe)
+
+        expected_ingredient_pairs = ingredient_pairs(recipe)
+
+        context = stub('context', cursor: nil)
+        actual_ingredient_pairs = ShoppingListTask.enumerator_builder
+          .enumerator(context: context).to_a
+
+        assert_equal expected_ingredient_pairs, actual_ingredient_pairs
+      end
+    end
+
+    test '#enumerator_builder.enumerator enumerates remaining ingredients for the cursor recipe' do
+      FancyRecipeAPI.fake_it_till_you_make_it do
+        expected_ingredient_pairs = ingredient_pairs(recipes(:vegan_tacos))
+        ingredients_so_far = expected_ingredient_pairs.shift(3)
+        cursor = ingredients_so_far.last.last
+
+        context = stub('context', cursor: cursor)
+        actual_ingredient_pairs = ShoppingListTask.enumerator_builder
+          .enumerator(context: context).to_a
+
+        assert_equal expected_ingredient_pairs, actual_ingredient_pairs
+      end
+    end
+
+    test '#count returns nil, as we cannot guess the recipe choice' do
+      assert_nil ShoppingListTask.count
+    end
+
+    private
+
+    def ingredient_pairs(recipe)
+      recipe.ingredients.map do |ingredient|
+        cursor = "#{recipe.id}:#{ingredient.id}"
+        [ingredient, cursor]
+      end
+    end
+  end
+end
+```
+
+Depending on its complexity, you may choose to test your Enumerator builder in
+isolation, in which case you can simplify the tests for your Task.
```

Comment:
I don't think we should go so much into details about how people should test their apps.